pkg/provisioner/ironic/ironic.go

```diff
@@ -785,12 +797,33 @@ func (p *ironicProvisioner) getUpdateOptsForNode(ironicNode *nodes.Node) (update
 		},
 	)
 
+	// boot_mode
+	_, ok := ironicNode.InstanceInfo["deploy_boot_mode"]
+	if !ok {
+		op = nodes.AddOp
+	} else {
+		op = nodes.ReplaceOp
+	}
```

Comment:
What's the plan for when we have a driver that doesn't use either UEFI or BIOS by default?

Reply:
Do you think we need to do more than extend the enum?

Reply:
Depends on whether the right thing to do for Ironic is to send an empty string or not to send anything.